controllers/machinedeletionremediation_controller.go

```diff
@@ -330,34 +339,53 @@ func (r *MachineDeletionRemediationReconciler) getMachine(remediation *v1alpha1.
 	return machine, nil
 }
 
-// saveMachineNameNs saves Machine Name and Namespace in a remediation's annotation
-func (r *MachineDeletionRemediationReconciler) saveMachineNameNs(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation, machineName, machineNamespace string) error {
+// saveMachineData saves target Machine data in a remediation's annotation
+func (r *MachineDeletionRemediationReconciler) saveMachineData(ctx context.Context, remediation *v1alpha1.MachineDeletionRemediation, machine *v1beta1.Machine) error {
 	annotations := remediation.GetAnnotations()
 	if annotations == nil {
 		annotations = make(map[string]string, 1)
-	} else if _, exists := annotations[MachineNameNsAnnotation]; exists {
-		return nil
 	}
 
-	annotations[MachineNameNsAnnotation] = fmt.Sprintf("%s/%s", machineNamespace, machineName)
+	if _, exists := annotations[MachineNameNsAnnotation]; !exists {
+		annotations[MachineNameNsAnnotation] =
+			fmt.Sprintf("%s/%s", machine.Namespace, machine.Name)
+	}
+
+	if _, exists := annotations[MachineOwnerAnnotation]; !exists {
+		owner, kind, err := getMachineOwnerReference(machine)
```

Comment:
nit:

variable name: `name` would be a better fit than `owner`
method name: you don't get the reference here, but name + kind